### PR TITLE
Rewrite Docker scenario guide

### LINKED ISSUE #72950: Rewrite Docker scenario guide
##### SUMMARY
The documentation says you need the `docker` pip package installed on the host running ansible. However, I do not have that package installed (`pip3 freeze | grep docker` returns nothing) but Ansible gives me an error when it's not installed on the server I try to provision. Therefore I am assuming that you need it on remote but not on the host. 

Documentation page for the issue is `https://docs.ansible.com/ansible/latest/scenario_guides/guide_docker.html`

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME

docker_*

##### ANSIBLE VERSION

```
ansible 2.10.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/natsu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

##### CONFIGURATION

```

```
(It is empty, no output)

##### OS / ENVIRONMENT

Ubuntu running inside WSL2, version is 20.04.1

##### ADDITIONAL INFORMATION

I think this could be caused by using Docker with the WSL2 integration. However, since Ansible tries to use the Python package and I ensured it really isn't installed inside my WSL2 instance (running `import docker` inside an interactive Python shell) I believe that this is more a bug in the documentation rather than having to do with said integration.

If helpful, I can also try to replicate this in an Ubuntu VM rather than WSL2.

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #72950.

I removed the link to the proposal repo (https://github.com/ansible/proposals/tree/master/docker), since it's probably terribly outdated.

CC @WojciechowskiPiotr

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/scenario_guides/guide_docker.rst

### CODE REVIEW

**Comment by felixfontein** [comment]
(For comparison, the current version: https://docs.ansible.com/ansible/latest/scenario_guides/guide_docker.html)

**Comment by felixfontein** [comment]
The undefined labels will start working once 1) ansible-collections/community.docker#61 is merged, 2) community.docker 1.1.0 is released, and 3) included in Ansible 2.10.5.

**Comment by WojciechowskiPiotr** [comment]
I think that the new guide covers the required essentials

**Review by bmillemathias — APPROVED** [review]

**Comment by felixfontein** [comment]
ansible-collections/community.docker#61 has been merged; I've updated this PR to use the new name of the inventory plugin added (community.docker.docker_containers instead of community.docker.docker).

**Comment by felixfontein** [comment]
community.docker 1.1.0 has been released. So once Ansible 2.10.5 has been released (in two days?) this PR should run through fine.

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 1]
```diff
@@ -1,112 +1,62 @@
...
-    the docker-py or docker python module. It is recommended to install the docker module if no support
-    for Python 2.6 is required. Please note that simply uninstalling one of the modules can leave the
-    other module in a broken state.
+.. code-block:: bash
 
-The docker_compose module also requires `docker-compose <https://github.com/docker/compose>`_
+    $ pip install 'docker-py>=1.10.0'
 
-.. code-block:: bash
+Please note that only one of ``docker`` and ``docker-py`` must be installed. Installing both will result in a broken installation. If this happens, Ansible will detect it and inform you about it. If that happens, you must uninstall both and reinstall the correct version.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Please install only one of ``docker`` or ``docker-py``. Installing both will result in a broken installation. If this happens, Ansible will detect it and inform you about it. If that happens, you must uninstall both and reinstall the correct version.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 2]
```diff
@@ -1,112 +1,62 @@
...
 
 
 Connecting to the Docker API
 ----------------------------
 
-You can connect to a local or remote API using parameters passed to each task or by setting environment variables.
-The order of precedence is command line parameters and then environment variables. If neither a command line
-option or an environment variable is found, a default value will be used. The default values are provided under
-`Parameters`_
+You can connect to a local or remote API using parameters passed to each task or by setting environment variables. The order of precedence is command line parameters and then environment variables. If neither a command line option or an environment variable is found, a default value will be used. The default values are provided under `Parameters`_.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
You can connect to a local or remote API using parameters passed to each task or by setting environment variables. The order of precedence is command line parameters and then environment variables. If neither a command line option nor an environment variable is found, Ansible uses the default value  provided under `Parameters`_.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 3]
```diff
@@ -1,112 +1,62 @@
...
 
-Control how modules connect to the Docker API by passing the following parameters:
+Most plugins and modules can be configured by the following parameters:
 
     docker_host
-        The URL or Unix socket path used to connect to the Docker API. Defaults to ``unix://var/run/docker.sock``.
-        To connect to a remote host, provide the TCP connection string. For example: ``tcp://192.0.2.23:2376``. If
-        TLS is used to encrypt the connection to the API, then the module will automatically replace 'tcp' in the
-        connection URL with 'https'.
+        The URL or Unix socket path used to connect to the Docker API. Defaults to ``unix://var/run/docker.sock``. To connect to a remote host, provide the TCP connection string. For example: ``tcp://192.0.2.23:2376``. If TLS is used to encrypt the connection to the API, then the module will automatically replace 'tcp' in the connection URL with 'https'.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The URL or Unix socket path used to connect to the Docker API. Defaults to ``unix://var/run/docker.sock``. To connect to a remote host, provide the TCP connection string (for example: ``tcp://192.0.2.23:2376``). If TLS is used to encrypt the connection to the API, then the module will automatically replace 'tcp' in the connection URL with 'https'.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 4]
```diff
@@ -155,176 +104,124 @@ running Ansible:
         Secure the connection to the API by using TLS and verify the authenticity of the Docker Host.
 
 
-Dynamic Inventory Script
-------------------------
-The inventory script generates dynamic inventory by making API requests to one or more Docker APIs. It's dynamic
-because the inventory is generated at run-time rather than being read from a static file. The script generates the
-inventory by connecting to one or many Docker APIs and inspecting the containers it finds at each API. Which APIs the
-script contacts can be defined using environment variables or a configuration file.
+Plain Docker Daemon: Images, Networks, Volumes and Containers
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Plain Docker daemon: images, networks, volumes, and containers
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 5]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
+-------------------------------------------------------------
 
-Groups
-......
-The script will create the following host groups:
+For working with a plain Docker Daemon, that is without Swarm, there are connection plugins, an inventory plugin, and several modules available:
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
For working with a plain Docker daemon, that is without Swarm, there are connection plugins, an inventory plugin, and several modules available:
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 6]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
- - container id
- - container name
- - container short id
- - image_name  (image_<image name>)
- - docker_host
- - running
- - stopped
+    docker connection plugin
+        The :ref:`community.docker.docker connection plugin <ansible_collections.community.docker.docker_connection>` allows to connect to Docker containers and execute modules in them using the Docker CLI utility. It essentially wraps ``docker exec`` and ``docker cp``. This connection plugin is supported by the :ref:`ansible.posix.synchronize module <ansible_collections.ansible.posix.synchronize_module>`.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker connection plugin <ansible_collections.community.docker.docker_connection>` uses the Docker CLI utility to connect to Docker containers and execute modules in them. It essentially wraps ``docker exec`` and ``docker cp``. This connection plugin is supported by the :ref:`ansible.posix.synchronize module <ansible_collections.ansible.posix.synchronize_module>`.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 7]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-Examples
-........
+    docker_api connection plugin
+        The :ref:`community.docker.docker_api connection plugin <ansible_collections.community.docker.docker_api_connection>` allows to connect to Docker containers and execute modules in them by talking directly to the Docker Daemon.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_api connection plugin <ansible_collections.community.docker.docker_api_connection>` talks directly to the Docker daemon to connect to Docker containers and execute modules in them.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 8]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-You can run the script interactively from the command line or pass it as the inventory to a playbook. Here are few
-examples to get you started:
+    docker_containers inventory plugin
+        The :ref:`community.docker.docker_containers inventory plugin <ansible_collections.community.docker.docker_containers_inventory>` allows to dynamically add Docker containers from a Docker Daemon to Ansible's inventory. See :ref:`dynamic_inventory` for details on dynamic inventories.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_containers inventory plugin <ansible_collections.community.docker.docker_containers_inventory>` allows you to dynamically add Docker containers from a Docker Daemon to your Ansible inventory. See :ref:`dynamic_inventory` for details on dynamic inventories.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 9]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-.. code-block:: bash
+        There is also a `docker inventory script <https://github.com/ansible-collections/community.general/blob/main/scripts/inventory/docker.py>`_. It is considered deprecated, please use the inventory plugin instead. The inventory plugin has several compatibility options. If you need to collect Docker containers from multiple Docker Daemons, you need to add every Docker Dameon as an individual inventory source.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The `docker inventory script <https://github.com/ansible-collections/community.general/blob/main/scripts/inventory/docker.py>`_ is deprecated. Please use the inventory plugin instead. The inventory plugin has several compatibility options. If you need to collect Docker containers from multiple Docker daemons, you need to add every Docker daemon as an individual inventory source.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 10]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    # Connect to the Docker API on localhost port 4243 and format the JSON output
-    DOCKER_HOST=tcp://localhost:4243 ./docker.py --pretty
+    docker_host_info module
+        The :ref:`community.docker.docker_host_info module <ansible_collections.community.docker.docker_host_info_module>` allows to retrieve information on a Docker Daemon, like all containers, images, volumes, networks and so on.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_host_info module <ansible_collections.community.docker.docker_host_info_module>` allows you to retrieve information on a Docker daemon, such as all containers, images, volumes, networks and so on.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 11]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    # Any container's ssh port exposed on 0.0.0.0 will be mapped to
-    # another IP address (where Ansible will attempt to connect via SSH)
-    DOCKER_DEFAULT_IP=192.0.2.5 ./docker.py --pretty
+    docker_login module
+        The :ref:`community.docker.docker_login module <ansible_collections.community.docker.docker_login_module>` allows to log in and out of a remote registry, like Docker Hub or a private registry. It provides similar functionality to the ``docker login`` and ``docker logout`` CLI commands.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_login module <ansible_collections.community.docker.docker_login_module>` allows you to log in and out of a remote registry, such as Docker Hub or a private registry. It provides similar functionality to the ``docker login`` and ``docker logout`` CLI commands.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 12]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    # Run as input to a playbook:
-    ansible-playbook -i ./docker.py docker_inventory_test.yml
+    docker_prune module
+        The :ref:`community.docker.docker_prune module <ansible_collections.community.docker.docker_prune_module>` allows to prune no longer needed containers, images, volumes and so on. It provides similar functionality to the ``docker prune`` CLI command.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_prune module <ansible_collections.community.docker.docker_prune_module>` allows  you to prune no longer needed containers, images, volumes and so on. It provides similar functionality to the ``docker prune`` CLI command.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 13]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
+        The :ref:`community.docker.docker_image module <ansible_collections.community.docker.docker_image_module>` provides full control over images, including: build, pull, push, tag and remove.
 
-        - name: Test docker_inventory, this will not connect to any hosts
-          hosts: all
-          gather_facts: no
-          tasks:
-            - debug:
-                msg: "Container - {{ inventory_hostname }}"
+    docker_image_info module
+        The :ref:`community.docker.docker_image_info module <ansible_collections.community.docker.docker_image_info_module>` allows to list and inspect images.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_image_info module <ansible_collections.community.docker.docker_image_info_module>` allows you to list and inspect images.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 14]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
-Configuration
-.............
-You can control the behavior of the inventory script by defining environment variables, or
-creating a docker.yml file (sample provided in https://raw.githubusercontent.com/ansible-collections/community.general/main/scripts/inventory/docker.py). The order of precedence is the docker.yml
-file and then environment variables.
+    docker_network module
+        The :ref:`community.docker.docker_network module <ansible_collections.community.docker.docker_network_module>` provides full control over Docker networks.
 
+    docker_network_info module
+        The :ref:`community.docker.docker_network_info module <ansible_collections.community.docker.docker_network_info_module>` allows to inspect Docker networks.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_network_info module <ansible_collections.community.docker.docker_network_info_module>` allows you to inspect Docker networks.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 15]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-Environment Variables
-;;;;;;;;;;;;;;;;;;;;;;
+    docker_volume_info module
+        The :ref:`community.docker.docker_volume_info module <ansible_collections.community.docker.docker_volume_info_module>` provides full control over Docker volumes.
 
-To connect to a single Docker API the following variables can be defined in the environment to control the connection
-options. These are the same environment variables used by the Docker modules.
+    docker_volume module
+        The :ref:`community.docker.docker_volume module <ansible_collections.community.docker.docker_volume_module>` allows to inspect Docker volumes.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_volume module <ansible_collections.community.docker.docker_volume_module>` allows you to inspect Docker volumes.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 16]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
-    DOCKER_HOST
-        The URL or Unix socket path used to connect to the Docker API. Defaults to unix://var/run/docker.sock.
+    docker_container module
+        The :ref:`community.docker.docker_container module <ansible_collections.community.docker.docker_container_module>` manages the container lifecycle by providing the ability to create, update, stop, start and destroy a Docker container.
 
-    DOCKER_API_VERSION:
-        The version of the Docker API running on the Docker Host. Defaults to the latest version of the API supported
-        by docker-py.
+    docker_container_info module
+        The :ref:`community.docker.docker_container_info module <ansible_collections.community.docker.docker_container_info_module>` allows to inspect a Docker container.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_container_info module <ansible_collections.community.docker.docker_container_info_module>` allows you to inspect a Docker container.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 17]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    DOCKER_TIMEOUT:
-        The maximum amount of time in seconds to wait on a response from the API. Defaults to 60 seconds.
 
-    DOCKER_TLS:
-        Secure the connection to the API by using TLS without verifying the authenticity of the Docker host server.
-        Defaults to False.
+Docker Compose
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst:157]:
```suggestion
Docker compose
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 18]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
+--------------
 
-    DOCKER_TLS_VERIFY:
-        Secure the connection to the API by using TLS and verifying the authenticity of the Docker host server.
-        Default is False
+The :ref:`community.docker.docker_compose module <ansible_collections.community.docker.docker_compose_module>`
+allows to use your existing Docker compose files to orchestrate containers on a single Docker daemon or on Swarm.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
allows you to use your existing Docker compose files to orchestrate containers on a single Docker daemon or on Swarm.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 19]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    DOCKER_SSL_VERSION:
-        Provide a valid SSL version number. Default value determined by docker-py, which at the time of this writing
-        was 1.0
+Docker Machine
+--------------
 
-In addition to the connection variables there are a couple variables used to control the execution and output of the
-script:
+The :ref:`community.docker.docker_machine inventory plugin <ansible_collections.community.docker.docker_machine_inventory>` allows to dynamically add Docker Machine hosts to Ansible's inventory.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
The :ref:`community.docker.docker_machine inventory plugin <ansible_collections.community.docker.docker_machine_inventory>` allows you to dynamically add Docker Machine hosts to your Ansible inventory.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 20]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    DOCKER_CONFIG_FILE
-        Path to the configuration file. Defaults to ./docker.yml.
 
-    DOCKER_PRIVATE_SSH_PORT:
-        The private port (container port) on which SSH is listening for connections. Defaults to 22.
+Docker Stack
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Docker stack
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 21]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
+------------
+
+The :ref:`community.docker.docker_stack module <ansible_collections.community.docker.docker_stack_module>` module allows to control Docker Stacks. Information on stacks can be retrieved by the :ref:`community.docker.docker_stack_info module <ansible_collections.community.docker.docker_stack_info_module>`, and information on stack tasks can be retrieved by the :ref:`community.docker.docker_stack_task_info module <ansible_collections.community.docker.docker_stack_task_info_module>`.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
The :ref:`community.docker.docker_stack module <ansible_collections.community.docker.docker_stack_module>` module allows you to control Docker stacks. Information on stacks can be retrieved by the :ref:`community.docker.docker_stack_info module <ansible_collections.community.docker.docker_stack_info_module>`, and information on stack tasks can be retrieved by the :ref:`community.docker.docker_stack_task_info module <ansible_collections.community.docker.docker_stack_task_info_module>`.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 22]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
+Docker Swarm
+------------
 
-Configuration File
-;;;;;;;;;;;;;;;;;;
+The community.docker collection provides multiple plugins and modules for managing Docker Swarms.
 
-Using a configuration file provides a means for defining a set of Docker APIs from which to build an inventory.
+Swarm Management
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Swarm management
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 23]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
+................
 
-The default name of the file is derived from the name of the inventory script. By default the script will look for
-basename of the script (in other words, docker) with an extension of '.yml'.
+One inventory plugin and several modules are provided to manage Docker Swarms:
 
-You can also override the default name of the script by defining DOCKER_CONFIG_FILE in the environment.
+    docker_swarm inventory plugin
+        The :ref:`community.docker.docker_swarm inventory plugin <ansible_collections.community.docker.docker_swarm_inventory>` allows to dynamically add all Docker Swarm nodes to Ansible's inventory.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_swarm inventory plugin <ansible_collections.community.docker.docker_swarm_inventory>` allows  you to dynamically add all Docker Swarm nodes to your Ansible inventory.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 24]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-Here's what you can define in docker_inventory.yml:
+    docker_swarm module
+        The :ref:`community.docker.docker_swarm module <ansible_collections.community.docker.docker_swarm_module>` allows to configure globally Docker Swarm manager nodes to join and leave swarms, and to change Docker Swarm configuration.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_swarm module <ansible_collections.community.docker.docker_swarm_module>` allows you to globally configure Docker Swarm manager nodes to join and leave swarms, and to change the Docker Swarm configuration.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 25]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    defaults
-        Defines a default connection. Defaults will be taken from this and applied to any values not provided
-        for a host defined in the hosts list.
+    docker_swarm_info module
+        The :ref:`community.docker.docker_swarm_info module <ansible_collections.community.docker.docker_swarm_info_module>` allows to retrieve information on Docker Swarm.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_swarm_info module <ansible_collections.community.docker.docker_swarm_info_module>` allows  you to retrieve information on Docker Swarm.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 26]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-    hosts
-        If you wish to get inventory from more than one Docker host, define a hosts list.
+    docker_node module
+        The :ref:`community.docker.docker_node module <ansible_collections.community.docker.docker_node_module>` allows to manage Docker Swarm nodes.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_node module <ansible_collections.community.docker.docker_node_module>` allows you to manage Docker Swarm nodes.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 27]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-For the default host and each host in the hosts list define the following attributes:
+    docker_node_info module
+        The :ref:`community.docker.docker_node_info module <ansible_collections.community.docker.docker_node_info_module>` allows to retrieve information on Docker Swarm nodes.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_node_info module <ansible_collections.community.docker.docker_node_info_module>` allows you to retrieve information on Docker Swarm nodes.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 28]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-.. code-block:: yaml
+Configuration Management
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Configuration management
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 29]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
-      description: The URL or Unix socket path used to connect to the Docker API.
-      required: yes
+The community.docker collection offers modules to manage Docker Swarm configurations and secrets:
 
-  tls:
-     description: Connect using TLS without verifying the authenticity of the Docker host server.
-     default: false
-     required: false
+    docker_config module
+        The :ref:`community.docker.docker_config module <ansible_collections.community.docker.docker_config_module>` allows to create and modify Docker Swarm configs.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_config module <ansible_collections.community.docker.docker_config_module>` allows to create and modify Docker Swarm configs.
```
```suggestion
        The :ref:`community.docker.docker_config module <ansible_collections.community.docker.docker_config_module>` allows you to create and modify Docker Swarm configs.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 30]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
-  tls_verify:
-     description: Connect using TLS without verifying the authenticity of the Docker host server.
-     default: false
-     required: false
+    docker_secret module
+        The :ref:`community.docker.docker_secret module <ansible_collections.community.docker.docker_secret_module>` allows to create and modify Docker Swarm secrets.
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
        The :ref:`community.docker.docker_secret module <ansible_collections.community.docker.docker_secret_module>` allows you to create and modify Docker Swarm secrets.
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 31]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
-  cert_path:
-     description: Path to the client's TLS certificate file.
-     default: null
-     required: false
 
-  cacert_path:
-     description: Use a CA certificate when performing server verification by providing the path to a CA certificate file.
-     default: null
-     required: false
+Swarm Services
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Swarm services
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 32]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
-  version:
-     description: The Docker API version.
-     required: false
-     default: will be supplied by the docker-py module.
 
-  timeout:
-     description: The amount of time in seconds to wait on an API response.
-     required: false
-     default: 60
+Helpful Links
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Helpful links
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 33]
```diff
@@ -118,19 +68,18 @@ Control how modules connect to the Docker API by passing the following parameter
...
-        to 'localhost'.
+        When verifying the authenticity of the Docker Host server, provide the expected name of the server. Defaults to ``localhost``.
 
     ssl_version
-        Provide a valid SSL version number. Default value determined by docker-py, which at the time of this writing
-        was 1.0
+        Provide a valid SSL version number. The default value is determined by the Docker SDK for Python.
 
 
 Environment Variables
```
Comment (samccann) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
Environment variables
```

**Review by samccann — COMMENTED** [review]
I changed the headings to use sentence case (that is -only the first word is capitalized). I left a couple of them, like Docker Swarm because the Docker docs have those capitalized.

File: docs/docsite/rst/scenario_guides/guide_docker.rst (hunk 17 above)
Comment (felixfontein) [reply docs/docsite/rst/scenario_guides/guide_docker.rst:157]: Docker Compose (with this capitalization) is the product name (https://en.wikipedia.org/wiki/Docker_(software)#Tools).

File: docs/docsite/rst/scenario_guides/guide_docker.rst [hunk 34]
```diff
@@ -155,176 +104,124 @@ running Ansible:
...
 
 
-Dynamic Inventory Script
-------------------------
-The inventory script generates dynamic inventory by making API requests to one or more Docker APIs. It's dynamic
-because the inventory is generated at run-time rather than being read from a static file. The script generates the
-inventory by connecting to one or many Docker APIs and inspecting the containers it finds at each API. Which APIs the
-script contacts can be defined using environment variables or a configuration file.
+Plain Docker daemon: images, networks, volumes, and containers
+-------------------------------------------------------------
```
Comment (felixfontein) [inline docs/docsite/rst/scenario_guides/guide_docker.rst]:
```suggestion
--------------------------------------------------------------
```

File: docs/docsite/rst/scenario_guides/guide_docker.rst (hunk 17 above)
Comment (samccann) [reply docs/docsite/rst/scenario_guides/guide_docker.rst:157]: good catch!

**Review by samccann — APPROVED** [review]

**Comment by felixfontein** [comment]
Restarting tests, since now they should pass.

**Comment by felixfontein** [comment]
CI is passing now (since 2.10.5 has been released and the docsite built).

**Comment by felixfontein** [comment]
@WojciechowskiPiotr @bmillemathias @samccann thanks a lot for reviewing this!!!

**Pull request merged**